src/nvim/window.c

```diff
@@ -2973,6 +2973,7 @@ int win_new_tabpage(int after)
     return FAIL;
   }
   curtab = newtp;
+  apply_autocmds(EVENT_TABNEW, arg, arg, false, curbuf);
 
   /* Create a new empty window. */
   if (win_alloc_firstwin(tp->tp_curwin) == OK) {
```

Comment:
(Note that I also switched `TabEnter` and `WinEnter` above. That order makes more sense to me.)